### PR TITLE
Issues/mapping extensions

### PR DESCRIPTION
**Pull request by AndreyDegtyarev**
Added method ToDto<> to facilitate converting entity to dto.

### CODE REVIEW

File: src/Nox.Reference.Data.Common/Extensions/MapperExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using Nox.Reference.Common;
+
+namespace Nox.Reference.Data.Common;
+
+public static class MapperExtensions
+{
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.Common/Extensions/MapperExtensions.cs:6]: Should we add "ToEntity" extension methods too?

File: src/Nox.Reference.Common/Extensions/MapperExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Reflection;
+using AutoMapper;
+
+namespace Nox.Reference.Common;
+
+public static class MapperHolder
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Common/Extensions/MapperExtensions.cs:6]: Why is this needed? I think we already have a mapper in World.Mapper field

File: src/Nox.Reference.Data.World/Helpers/World.cs [hunk 3]
```diff
@@ -1,21 +1,17 @@
...
-using Microsoft.Extensions.DependencyInjection;
+﻿using Microsoft.Extensions.DependencyInjection;
 
 namespace Nox.Reference.Data.World;
 
 public static class World
 {
     private static readonly IServiceProvider _serviceProvider;
-    // TODO: possibly add specialized flat mapping methods per entity
-    public static readonly IMapper Mapper;
```
Comment (ILoveMyTortoise) [inline src/Nox.Reference.Data.World/Helpers/World.cs:10]: Why remove it? Let's discuss it

File: tests/Nox.Reference.Data.Machine.Tests/MacAddresses.Tests.cs [hunk 4]
```diff
@@ -57,7 +63,21 @@ public void Setup()
...
+    }
+
+    [Test]
+    public void GetVendorMacAddress_ConvertToDto_ReturnsSuccess()
+    {
+        MacAddress macAddress = Machine.Machine.MacAddresses.Get("00:16:F6:11:22:33")!;
+
+        MacAddressInfo macAddressInfo = macAddress.ToDto<MacAddressInfo>();
+
+        Assert.That(macAddressInfo, Is.Not.Null);
```
Comment (ILoveMyTortoise) [inline tests/Nox.Reference.Data.Machine.Tests/MacAddresses.Tests.cs:80]: why not assert.multiple?

File: src/Nox.Reference.Data.Common/Extensions/MapperExtensions.cs (hunk 1 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Data.Common/Extensions/MapperExtensions.cs:6]: I don't think this is good idea. 
1) Since ToEntity() is extension method it will be applied for any object in a solution 
2) We need dto to entity converting only when we write data internaly

File: src/Nox.Reference.Common/Extensions/MapperExtensions.cs (hunk 2 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Common/Extensions/MapperExtensions.cs:6]: MapperHolder serves to register all included mapping. 
Because we use static instance of mapper we should register its mapping.

File: src/Nox.Reference.Data.World/Helpers/World.cs (hunk 3 above)
Comment (AndreyDegtyarev) [reply src/Nox.Reference.Data.World/Helpers/World.cs:10]: Since MapperHolder is used

File: src/Nox.Reference.Common/Extensions/MapperExtensions.cs (hunk 2 above)
Comment (ILoveMyTortoise) [reply src/Nox.Reference.Common/Extensions/MapperExtensions.cs:6]: let's discuss it

File: tests/Nox.Reference.Data.Machine.Tests/MacAddresses.Tests.cs [hunk 5]
```diff
@@ -34,15 +35,20 @@ public void Setup()
...
-        Assert.That(info, Is.Not.Null);
-        Assert.That(expectedPrefix, Is.EqualTo(info.Id));
-
+        var info = _macAddressContext.MacAddresses.Get(input)!;
+        Assert.Multiple(() =>
+        {
+            Assert.That(info, Is.Not.Null);
+            Assert.That(expectedPrefix, Is.EqualTo(info.Id));
+        });
         var mappedInfo = Machine.Machine.Mapper.Map<MacAddressInfo>(info);
```
Comment (ILoveMyTortoise) [inline tests/Nox.Reference.Data.Machine.Tests/MacAddresses.Tests.cs:44]: edit here if you deleting it

**Review by ILoveMyTortoise — APPROVED** [review]
